### PR TITLE
[MBL-17274][Student] Remind me UI tests

### PR DESCRIPTION
**Pull request by kristofnemere**
Test plan: -

refs: MBL-17274
affects: Student
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by tamaskozmer — APPROVED** [review]

File: apps/student/src/androidTest/java/com/instructure/student/ui/interaction/AssignmentDetailsInteractionTest.kt [hunk 1]
```diff
@@ -385,6 +387,130 @@ class AssignmentDetailsInteractionTest : StudentTest() {
...
+        val assignment = data.addAssignment(course.id, name = "Test Assignment", dueAt = Calendar.getInstance().apply {
+            add(Calendar.DAY_OF_MONTH, 1)
+        }.time.toApiString())
+        goToAssignmentList()
+
+        assignmentListPage.clickAssignment(assignment)
+        assignmentDetailsPage.clickAddReminder()
+        assignmentDetailsPage.clickCustom()
+        assignmentDetailsPage.fillQuantity("15")
+        assignmentDetailsPage.clickHoursBefore()
```
Comment (kdeakinstructure) [inline apps/student/src/androidTest/java/com/instructure/student/ui/interaction/AssignmentDetailsInteractionTest.kt:472]: I would put an assert here that until this point the 'Done' button is disabled.

File: apps/student/src/androidTest/java/com/instructure/student/ui/interaction/AssignmentDetailsInteractionTest.kt [hunk 2]
```diff
@@ -385,6 +387,130 @@ class AssignmentDetailsInteractionTest : StudentTest() {
...
+        }.time.toApiString())
+        goToAssignmentList()
+
+        assignmentListPage.clickAssignment(assignment)
+
+        assignmentDetailsPage.assertReminderSectionDisplayed()
+    }
+
+    @Test
+    @TestMetaData(Priority.IMPORTANT, FeatureCategory.ASSIGNMENTS, TestCategory.INTERACTION)
```
Comment (kdeakinstructure) [inline apps/student/src/androidTest/java/com/instructure/student/ui/interaction/AssignmentDetailsInteractionTest.kt:421]: testAddReminder, testRemoveReminder, testAddCustomReminder tests are mandatory priority according to this feature.

File: apps/student/src/androidTest/java/com/instructure/student/ui/pages/AssignmentDetailsPage.kt [hunk 3]
```diff
@@ -239,6 +246,68 @@ open class AssignmentDetailsPage : BasePage(R.id.assignmentDetailsPage) {
...
+    }
+
+    fun removeReminderWithText(text: String) {
+        onView(
+            allOf(
+                withId(R.id.remove),
+                hasSibling(withText(text))
+            )
+        ).click()
+        onView(withText(R.string.yes)).scrollTo().click()
```
Comment (kdeakinstructure) [inline apps/student/src/androidTest/java/com/instructure/student/ui/pages/AssignmentDetailsPage.kt:288]: Isn't it the android.R.string.yes resource on the dialog?

File: apps/student/src/androidTest/java/com/instructure/student/ui/pages/AssignmentDetailsPage.kt [hunk 4]
```diff
@@ -239,6 +246,68 @@ open class AssignmentDetailsPage : BasePage(R.id.assignmentDetailsPage) {
...
+    }
+
+    fun assertReminderNotDisplayedWithText(text: String) {
+        onView(withText(text)).check(doesNotExist())
+    }
+
+    fun clickCustom() {
+        onData(anything()).inRoot(isDialog()).atPosition(6).perform(click())
```
Comment (kdeakinstructure) [inline apps/student/src/androidTest/java/com/instructure/student/ui/pages/AssignmentDetailsPage.kt:296]: Nice work! 👍

**Review by kdeakinstructure — CHANGES_REQUESTED** [review]
See comments.

File: apps/student/src/androidTest/java/com/instructure/student/ui/pages/AssignmentDetailsPage.kt (hunk 3 above)
Comment (kristofnemere) [reply apps/student/src/androidTest/java/com/instructure/student/ui/pages/AssignmentDetailsPage.kt:288]: We use our own on several places.

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Pull request merged**